controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -927,6 +936,18 @@ func (r *BareMetalHostReconciler) actionInspecting(prov provisioner.Provisioner,
 
 	clearError(info.host)
 	info.host.Status.HardwareDetails = details
+
+	if !info.host.DeletionTimestamp.IsZero() {
+		info.log.Info(fmt.Sprintf("will not attempt to create hardwareData in %q", hd.Namespace))
+		return actionComplete{}
```

Comment:
This change is not needed, it's adding an earlier return but the code it was preventing execution of doesn't exist in this branch.